File: airport_and_city_search/v1/locations/java/AirportCitySearch.java
package examples.media.files;

import com.amadeus.Amadeus;
import com.amadeus.Params;
import com.amadeus.exceptions.ResponseException;
import com.amadeus.resources.FlightDate;
import com.amadeus.resources.Location;

public class AirportCitySearch {

  public static void main(String[] args) throws ResponseException {

    Amadeus amadeus = Amadeus
        .builder("YOUR_API_ID","YOUR_API_SECRET")
        .build();

    // Airport & City Search (autocomplete)
    // Find all the cities and airports starting by the keyword 'LON'
    Location[] locations = amadeus.referenceData.locations.get(Params
      .with("keyword", "LON")
      .and("subType", Locations.ANY));

    if(locations[0].getResponse().getStatusCode() != 200) {
        System.out.println("Wrong status code: " + locations[0].getResponse().getStatusCode());
        System.exit(-1);
    }
    System.out.println(locations[0]);
  }
}

File: airport_and_city_search/v1/locations/iOS SDK/AirportCitySearch.swift
import Amadeus

var amadeus: Amadeus = Amadeus(
    client_id: "YOUR_API_ID",
    client_secret: "YOUR_API_SECRET"
)

/// Airport & City Search (autocomplete)
/// Find all the cities and airports starting by the keyword 'LON'
amadeus.referenceData.locations.get(params: ["subType": "AIRPORT,CITY",
                                             "keyword": "LON"],
                                    onCompletion: {
                                        response, error in
                                        if error == nil {
                                            print(response!.data)
                                        }
                                        })
